app/jobs/risc_delivery_job.rb

```diff
@@ -49,8 +49,10 @@ def perform(
       )
     end
   rescue *NETWORK_ERRORS, RedisRateLimiter::LimitError => err
-    raise err if !inline?
-
+    if self.exceptions < 2
+      raise err if !inline?
+    end
```

Comment:
This is great! The network errors and retry limits have different retry limits, and probably need different logic. I'm not sure on the behavior of if there have been retries for both, but I'm not sure if we need to figure that out here?

This is my attempt, but there's probably a better way:

```suggestion
   raise err if NETWORK_ERRORS.any? { |e| err.kind_of?(e) } && self.exceptions < 2
   raise err if err.kind_of?(RedisRateLimiter::LimitError) && self.exceptions < 10
```

Reply:
I think this could work. My only question is do we need the "if !inline?" check?

Also, I'm wondering your thoughts if it would be cleaner if we just had two raises, one for each kind of exception type.

Reply:
Thanks for catching that, yeah, we should keep it. Two raises as in a separate line for each makes sense to me.

Reply:
Ok let me make the appropriate PR updates.